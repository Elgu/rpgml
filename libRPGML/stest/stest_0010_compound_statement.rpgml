PRINT( Frame{ int32 i = 23; string foo = "foo"; double d = 0.125; } );
PRINT( "\n" );

int i = 0;
PRINT( i );
PRINT( "\n" );

{
  i = 13;
  PRINT( i );
  PRINT( "\n" );

  int i = 42;
  PRINT( i );
  PRINT( "\n" );

  i = 23;
  PRINT( i );
  PRINT( "\n" );
}

PRINT( i );
PRINT( "\n" );

i = 53;
PRINT( i );
PRINT( "\n" );

PRINT(
  Frame{ 
    int i = 73;

    PRINT( i );
    PRINT( "\n" );

    Method set_i( new_i )
    {
      PRINT( i );
      PRINT( "\n" );
      i = new_i;
      PRINT( i );
      PRINT( "\n" );
      return this;
    }

    Method get_i()
    {
      PRINT( i );
      PRINT( "\n" );
      return i;
    }
  }.set_i( 93 ).get_i()
  );

