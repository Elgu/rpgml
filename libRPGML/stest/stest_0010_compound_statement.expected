Frame{  int32 i = 23; string foo = "foo"; double d = 0.125; }
0
13
42
23
13
53
73
73
93
93
93
